parlai/core/metrics.py

```diff
@@ -523,6 +529,60 @@ def compute_many(
         )
 
 
+class IntraDistinctMetric(AverageMetric):
+    """
+    Compute intra-distinct (per-utterance).
+    """
+
+    @classmethod
+    def _ngram(cls, seq, n: int):
```

Comment:
this looks like this NGramIterator for unlikelihood training: https://github.com/fairinternal/ParlAI-Internal/blob/master/projects/meena/safety/unlikelihood/agents.py#L33-L53 maybe we can extract those a separable feature?

Reply:
I debated. I even have it copy pasted here. OTOH it's two lines of code lol